### ISSUE TITLE
Early request not intercepted in Firefox for most recent msw version

### ISSUE THREAD

**Issue by celcius112** [kettanaito/msw#1090]
## Environment

| Name    | Version |
| ------- | ------- |
| msw     | 0.36.8   |
| browser | Firefox  96  |
| OS      | MacOS 12.1     |

First of all, thank you for this very handy library :)

## Current behavior

Updating the msw version to 0.36.8 in the Angular example project (https://github.com/mswjs/examples/tree/master/examples/rest-angular) breaks the request interception in Firefox for early requests.

Here's a zip containing : 
- An upgrade to msw 0.36.8
- A modification in AppComponent to perform an early request to `login`
[examples.zip](https://github.com/mswjs/msw/files/8031346/examples.zip)

Launching the Angular application for the first time after a `yarn start` will result with a 404 being thrown. Reloading the page or clicking the submit button will make it work.
<img width="964" alt="Screenshot 2022-02-09 at 10 42 23" src="https://user-images.githubusercontent.com/1503748/153169861-e220040c-933c-4785-ae2b-0c86cf743db4.png">

**Comment by kettanaito** [comment]
Hey, @celcius112. Thanks for reporting this. 

I confirm this is still an issue on Firefox only even with `0.39.1`. The recent worker activation Promise fixes we've published do not fix this. Needs investigation.

**Comment by kettanaito** [comment]
This is related to #220. Firefox does not expose `XMLHttpRequest` to Service Worker, so the worker doesn't trigger its "fetch" event when such requests occur. This is a known limitation of Firefox and there's nothing we can do about it. 

I've ensured that the fetch/XMLHttpRequest requests are deferred correctly and they are. Requests are performed only once the worker is active. It still cannot handle the XMLHttpRequest because Firefox doesn't notify the worker about such requests. It does work as expected when using `window.fetch` upon initial load (early request). 

```ts
  ngOnInit(): void {
    fetch('/login', {
      method: 'POST',
      headers: { 'content-type': 'application/json' },
      body: JSON.stringify({ username: 'test' }),
    })
  }
```

I understand this may be not the way to request resources in Angular but overall you must consider using `window.fetch`. I suspect that `this.http.post` used in `this.userService.login` still creates an XHR instead. May be a dependency issue, may be a framework consensus.

Either way, MSW cannot (and should not) circumvent browser limitations. Being a developer tool, you can use MSW in other browsers instead.

**Comment by celcius112** [comment]
Hello @kettanaito, thank you for your answer. 

What I find weird is that a retry make it work correctly (for instance with a `retryWhen` https://www.learnrxjs.io/learn-rxjs/operators/error_handling/retrywhen). I do not mind retrying since I'm already retrying for other reasons in the local environment :).

**Comment by followben** [comment]
Thanks for the writeup @kettanaito - would you accept a PR to highlight this issue in the docs (possibly as an aside in the [Deferred Mounting](https://mswjs.io/docs/recipes/deferred-mounting) page)? I just spent a good couple of hours trying to figure out why deferred mounting wouldn't resolve my timing issue with axios and firefox and auth calls on mount before I found this by searching closed GH issues :) Feel it'd be good to highlight the shortcomings of Firefox without fetch.

**Comment by kettanaito** [comment]
Hey, @followben. I was thinking we should have a "Limitations" page in the docs. This particular Firefox limitation would rest there nicely. I've already written this page for the new version of the docs, but if you feel this is pressing, you can add a copy in a pull request!

**Comment by kettanaito** [comment]
@celcius112, then I suspect the retry mechanism employs a different way to fetch resources. Different than XHR.

**Comment by followben** [comment]
@kettanaito FYI I switched from react-scripts (CRA) to Vite overnight and deferred mounting now works nicely with Firefox and Axios. Not sure if that's just because Vite's dev server starts up so quickly or what, but I'll take it :) Limitations page sounds good... is that the [mswjs.io](https://github.com/mswjs/mswjs.io) repo? Which branch would I find it in?

**Comment by kettanaito** [comment]
@followben, yes, that's the correct repo. There's currently no such page in this version of the docs, so you may add one at the same level as the "FAQ" page.
